Intersect.Client/Maps/MapInstance.cs

```diff
@@ -1003,104 +1003,63 @@ private GameTileBuffer[][] DrawMapLayer(string layerName, float xoffset = 0, flo
+            mCurFogIntensity = Id == Globals.Me.MapId
+                ? Math.Min(1, mCurFogIntensity + elapsedTime / 2000f)
+                : Math.Max(0, mCurFogIntensity - elapsedTime / 2000f);
 
-                    if (mFogCurrentX < fogTex.GetWidth())
-                    {
-                        mFogCurrentX += fogTex.GetWidth();
-                    }
+            // Calculate the number of times the fog texture needs to be drawn to cover the map area.
+            var xCount = Options.MapWidth * Options.TileWidth * 3 / fogTex.GetWidth();
+            var yCount = Options.MapHeight * Options.TileHeight * 3 / fogTex.GetHeight();
 
-                    if (mFogCurrentX > fogTex.GetWidth())
-                    {
-                        mFogCurrentX -= fogTex.GetWidth();
-                    }
-
-                    if (mFogCurrentY < fogTex.GetHeight())
-                    {
-                        mFogCurrentY += fogTex.GetHeight();
-                    }
+            // Update the fog texture's position based on its speed and elapsed time.
+            mFogCurrentX += elapsedTime / 1000f * FogXSpeed * 2;
+            mFogCurrentY += elapsedTime / 1000f * FogYSpeed * 2;
 
-                    if (mFogCurrentY > fogTex.GetHeight())
-                    {
-                        mFogCurrentY -= fogTex.GetHeight();
-                    }
+            // Handle cases where the fog texture's position goes out of bounds.
+            mFogCurrentX %= fogTex.GetWidth();
+            mFogCurrentY %= fogTex.GetHeight();
 
-                    var drawX = (float)Math.Round(mFogCurrentX);
-                    var drawY = (float)Math.Round(mFogCurrentY);
+            // Round the fog texture's position to the nearest integer value.
+            var drawX = (float)Math.Round(mFogCurrentX);
+            var drawY = (float)Math.Round(mFogCurrentY);
 
-                    for (var x = -1; x < xCount; x++)
-                    {
-                        for (var y = -1; y < yCount; y++)
-                        {
-                            var fogW = fogTex.GetWidth();
-                            var fogH = fogTex.GetHeight();
-                            Graphics.DrawGameTexture(
-                                fogTex, new FloatRect(0, 0, fogW, fogH),
-                                new FloatRect(
-                                    GetX() - Options.MapWidth * Options.TileWidth * 1f + x * fogW + drawX,
-                                    GetY() - Options.MapHeight * Options.TileHeight * 1f + y * fogH + drawY, fogW, fogH
-                                ), new Intersect.Color((byte)(FogTransparency * mCurFogIntensity), 255, 255, 255),
-                                null, GameBlendModes.None
-                            );
-                        }
-                    }
+            for (var x = -1; x < xCount; x++)
+            {
+                for (var y = -1; y < yCount; y++)
```

Comment:
Why are these initialized to -1 when they are still `< Count`? What are the effects of changing it to start with `0` or end with `<= Count`?

Reply:
it seems to have the same outcome in game 🤔 i tho that X and y are initialized to -1 because the loop will run one extra time for each of them due to the < operator in the loop condition. This means that the inner loop should run an extra time for each outer loop iteration so it effectively drawing an extra row and column of fog texture. but in the practice, it seems to have no apparent change to set it to 0.

Should i change this and the one present within the editor to 0 then (?)

Reply:
changed them to the following on both ends:

![imagen](https://user-images.githubusercontent.com/17498701/210117192-ed9aac02-a889-4d01-8064-df63f1acf210.png)

**but within the editor:** there's some unwanted outcome :

![imagen](https://user-images.githubusercontent.com/17498701/210117241-c005951c-014d-4ba3-ac68-e8e7b1e935f7.png)

![imagen](https://user-images.githubusercontent.com/17498701/210117247-44790c9c-7d80-4a08-aa1a-4c7652cb8dda.png)

i will leave them as they were originally for now.

Reply:
just figured out ~